File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* middleNode(struct ListNode* head) {
    struct ListNode* temp;
    temp=head;
    int len=0,i=0;
    while(temp!=NULL){
        len++;
        temp=temp->next;
    }
    temp=head;
    while(i<(len/2)){
        temp=temp->next;
        i++;
    }
    return temp;

}

File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
def leng(head,count):
        temp=head
        while temp.next is not None:
            temp=temp.next
            count+=1
        return count
        
class Solution(object):
    def middleNode(self, head):
        count=1
        l=leng(head,count)
        curr=head
        mid=l//2
        if l%2==0:
            while mid!=0:
                curr=curr.next
                mid-=1
            return curr
        else:
            while mid!=0:
                curr=curr.next
                mid-=1
            return curr
    